torch/csrc/dynamo/eval_frame.c

```diff
@@ -373,6 +378,8 @@ static PyObject* _custom_eval_frame(
   }
 
   // TODO(jansel): investigate directly using the "fast" representation
+  // TODO(alband): This is WRONG for python3.11+ we pass in a _PyInterpreterFrame
```

Comment:
This is problematic: the frame we're getting doesn't have a PyFrameObject associated with it (and the default eval actually asserts it is still empty so we don't want to populate it here).
Even if we were creating it here, the PyFrameObject we would get would not be initialized and so most PyFrame_* APIs will raise an error anyway due to the newly added checks that frames must be complete before calling these APIs.

I tested locally that calling directly https://github.com/python/cpython/blob/d98ca8172c39326bb200308a5191ceeb4a262d53/Objects/frameobject.c#L1182 does work fine here. I need to find a way to call it from here without massive amount of copy pasting.

This is also left for a follow up PR to keep this one simple in case this is a lot of work.

Reply:
This is done one PR above this in the stack. The copy-paste is not the best but works.